staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -28,12 +28,26 @@ import (
 	"time"
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog/v2"
 )
 
-// New returns an http.RoundTripper that will provide the authentication
-// or transport level security defined by the provided Config.
+// DialerStopCh is stop channel that is passed down to dynamic cert dialer.
+// It's exposed as variable for testing purposes to avoid testing for goroutine
+// leakages.
+var DialerStopCh = wait.NeverStop
+
+// New returns an http.RoundTripper with TLS config bound with wait.NeverStop channel
 func New(config *Config) (http.RoundTripper, error) {
+	return NewWithContext(DialerStopCh, config)
+}
+
+// NewWithContext returns an http.RoundTripper that will provide the authentication
+// or transport level security defined by the provided Config bounded with a channel.
+
+// channel is used to determine when to stop reloading certs. Used to avoid goroutine leaks caused from
+// ensuring to reload peer certs needed for communication between apiservers.
+func NewWithContext(ch <-chan struct{}, config *Config) (http.RoundTripper, error) {
```

Comment:
this does not look right, why do we need this?

Reply:
oh without this, I was seeing [unexpected goroutines] error possibly coming from (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/cache.go#L122) at the end of my integration test.

Reply:
did not infact need this, cleaned it up.